ppa/archive/management/commands/hathi_rsync.py

```diff
@@ -26,22 +30,64 @@ def handle(self, *args, **kwargs):
         # use ids specified via command line when present
         htids = kwargs.get("htids", [])
 
-        # if hathi ids not specified via command line,
-        # get all non-suppressed hathi records
-        if not htids:
-            htids = DigitizedWork.objects.filter(
-                status=DigitizedWork.PUBLIC, source=DigitizedWork.HATHI
-            ).values_list("source_id", flat=True)
+        # by default, sync data for all non-suppressed hathi source ids
+        digworks = DigitizedWork.objects.filter(
+            status=DigitizedWork.PUBLIC, source=DigitizedWork.HATHI
+        )
 
-        # NOTE: if htid is specified, should we verify that it's
-        # in the db and not suppressed? (should import first if not)
+        # if htids are specified via parameter, use them to filter
+        # the queryset, to ensure we only sync records that are
+        # in the database and not suppressed
+        if htids:
+            digworks = digworks.filter(source_id__in=htids)
+        # NOTE: report here on any skipped ids?
 
-        self.stdout.write(
-            self.style.SUCCESS("Synchronizing data for %d records" % len(htids))
-        )
-        # even if verbosity is zero we want an output file
+        # generate a list of unique source ids from the queryset
+        hathi_ids = digworks.values_list("source_id", flat=True).distinct()
+        self.stdout.write("Synchronizing data for %d records" % len(hathi_ids))
+        # we always want itemized rsync output, so we can report
+        # on which volumes were updated
         htimporter = HathiImporter(
-            source_ids=htids, rsync_output=self.verbosity or True
+            source_ids=hathi_ids, rsync_output=True, output_dir="/tmp"
```

Comment:
Is `/tmp` a good output directory for this?

Reply:
Oh, I meant to look into that and forgot. If and when we set this functionality to run as a cron job, that's not a good place. In staging/production, maybe it could go into the logging directory. It might be good enough for now, while we're assessing.